### PR TITLE
Test parallel circleci test

### PR DESCRIPTION
**Pull request by samuelhwilliams**
### Description of change
Run our tests and linting in parallel so that tests complete faster and linting failures are reported sooner.

### Checklist

* [ ] Have tests been added to cover any changes?
* [ ] Has the [CHANGELOG](https://github.com/uktrade/data-workspace/blob/master/CHANGELOG.md) been updated?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+  docker-test:
     docker:
-      # Test and code run in _other_ docker containers, so mostly doesn't matter
       - image: circleci/python:3.7
     steps:
       - checkout
       - setup_remote_docker
       - run:
-          name: Run tests
+          name: Run test
```
Comment (allait) [inline .circleci/config.yml]: There is more than 1 though 😄

Comment (samuelhwilliams) [reply .circleci/config.yml]: Typo 🤷‍♂️

**Comment by samuelhwilliams** [comment]
I'm prepared to be wrong, but I'd be surprised if it has a hard requirement on 2 before starting any. I think it's more likely that, if just one is free, it will start doing one of the jobs. And when another comes free, it will then do the second. In which case we don't really lose anything.

This mostly came from wanting to have linting come back before the tests are complete. We could reorder them and run linting first, but I don't think it really gets us anything - if I'm right with the assumption above.

**Comment by samuelhwilliams** [comment]
Linting is quick yes, but with dependencies it's not instant (30-60s). I think it's fine to run in parallel.

I'm not sure how it's slower to procure two than one if they're procuring in parallel.

**Comment by samuelhwilliams** [comment]
Also it's nice to see the separate github status checks for linting and tests 🙈😂

**Pull request merged**